receiver/awscontainerinsightreceiver/internal/neuron/pod_attribute_decorator.go

```diff
@@ -0,0 +1,131 @@
+			neuronCoresPerDevice, _ := strconv.Atoi(neuronCoresPerDeviceValue.AsString())
+
+			for k := 0; k < metrics.Len(); k++ {
+				m := metrics.At(k)
+				pdc.AddPodCorrelationAttributes(getMetricDatapoints(m), neuronCoresPerDevice)
+			}
+		}
+	}
+	return md
+}
+
+func getMetricDatapoints(m pmetric.Metric) pmetric.NumberDataPointSlice {
+	switch m.Type() {
+	case pmetric.MetricTypeGauge:
+		return m.Gauge().DataPoints()
+	case pmetric.MetricTypeSum:
+		return m.Sum().DataPoints()
+	default:
+		return pmetric.NewNumberDataPointSlice()
+	}
+}
+
+func (pdc *PodAttributesDecoratorConsumer) AddPodCorrelationAttributes(metricDatapoints pmetric.NumberDataPointSlice, neuronCoresPerDevice int) {
+	for i := 0; i < metricDatapoints.Len(); i++ {
+		attributes := metricDatapoints.At(i).Attributes()
+		var containerInfo *stores.ContainerInfo
+
+		if neuronDeviceIndex, neuronDeviceIndexPresent := attributes.Get(neuronDeviceAttributeKey); neuronDeviceIndexPresent {
+			// get container info from neuronDeviceIndex
+			neuronDeviceIndex := neuronDeviceIndex.AsString()
+			if neuronDeviceIndexPresent {
+				containerInfo = pdc.getContainerInfoForNueronDeviceIndex(neuronDeviceIndex)
+
+			}
+		} else if neuronCoreIndex, neuronCoreIndexPresent := attributes.Get(neuronCoreAttributeKey); neuronCoreIndexPresent {
+			// get container info from neuronCore
+			containerInfo = pdc.PodResourcesStore.GetContainerInfo(neuronCoreIndex.AsString(), neuronCoreResourceName)
+			neuronDeviceIndex := getNeuronDeviceIndexFromCoreAttribute(neuronCoreIndex, neuronCoresPerDevice)
+			if containerInfo == nil {
+				// else get container info from calculated neuronDeviceIndex
+				containerInfo = pdc.getContainerInfoForNueronDeviceIndex(neuronDeviceIndex)
+			}
+			attributes.PutStr(neuronDeviceAttributeKey, neuronDeviceIndex)
+		}
+		populateAttributes(&attributes, containerInfo)
+	}
+}
+
+func (pdc *PodAttributesDecoratorConsumer) getContainerInfoForNueronDeviceIndex(neuronDeviceIndex string) *stores.ContainerInfo {
+	containerInfo := pdc.PodResourcesStore.GetContainerInfo(neuronDeviceIndex, neuronDeviceResourceName)
+	if containerInfo == nil {
+		// Alt resource name is to support backward compatibility in neuron monitor : https://awsdocs-neuron.readthedocs-hosted.com/en/latest/containers/tutorials/k8s-setup.html
+		containerInfo = pdc.PodResourcesStore.GetContainerInfo(neuronDeviceIndex, neuronDeviceResourceNameAlt)
+	}
+	return containerInfo
+}
+
+func populateAttributes(attributes *pcommon.Map, containerInfo *stores.ContainerInfo) {
+	if containerInfo != nil {
+		attributes.PutStr(ci.AttributeContainerName, containerInfo.ContainerName)
+		attributes.PutStr(ci.AttributeK8sPodName, containerInfo.PodName)
+		attributes.PutStr(ci.AttributePodName, containerInfo.PodName)
+		attributes.PutStr(ci.AttributeK8sNamespace, containerInfo.Namespace)
+		attributes.PutStr(ci.AttributeFullPodName, containerInfo.PodName+"."+containerInfo.Namespace)
```

Comment:
That's not the right logic for FullPodName, but also the k8sdecorator will add it for you anyway so you can remove from here.